packages/filebrowser-extension/src/index.ts

```diff
@@ -1300,10 +1302,17 @@ namespace Private {
         // Restore the model without populating it.
         await browser.model.restore(browser.id, false);
         if (paths.file) {
-          await commands.execute(CommandIDs.openPath, {
+          let widget = await commands.execute(CommandIDs.openPath, {
             path: paths.file,
             dontShowBrowser: true
           });
+          if (widget !== undefined) {
+            for (let w of app.shell.widgets('main')) {
+              if ((w as MainAreaWidget).content instanceof Launcher) {
```

Comment:
Maybe we should add a command in `launcher-extension` to close all launcher instances instead of doing this here. What do you think?

Reply:
Yes, it would be better than the current implementation.

But I think a better solution could be to not open the Launcher at all before the main area is fully restored (including the file from URL).
Currently both the Launcher extension and the document from URL are awaiting the same promise `browser.restored()`. Maybe the creation of the Launcher could move from `launcher-extension` to `filebrowser-extension`. then we could wait for `browser.restored()` -> opening file from URL -> and finally opening a launcher if there is no widget in the main area.

Let me know which solution is better, I can add it to this PR.

Reply:
Hm, both of those sound like good solutions. Avoiding the creation of a launcher when it's not necessary is more efficient and performant.

Reply:
In fact I think  this is now useless and this part can be removed.

It was useful when:
- the saved workspace was empty
- we start in `doc` mode with a document in URL (a `Launcher` was also created)
- then we switch to `lab` mode, the opened document remains, with the launcher

Currently the opened document is closed when we switch back to `lab` mode, so:
- if the saved workspace was empty we need a `Launcher`
- if the saved workspace was not empty it is restored without the `Launcher`

If we remove it, when starting in `doc` mode there will be 2 widgets in main area (the document and a `Launcher`), but only the document will be visible...

Do you think we can remove it ?

Reply:
> Hm, both of those sound like good solutions. Avoiding the creation of a launcher when it's not necessary is more efficient and performant.

OK, thanks for your reply, I can handle it.